torch/_functorch/aot_autograd.py

```diff
@@ -2180,7 +2180,8 @@ def forward(ctx, *deduped_flat_tensor_args):
                 assert all(
                     [isinstance(x, torch.Tensor) for x in tensors_saved_for_backwards]
                 )
-                ctx.save_for_backward(*tensors_saved_for_backwards)
+                # See Note [Detaching saved tensors in AOTAutograd]
+                ctx.save_for_backward(*map(lambda x: x.detach() if x._is_view() else x, tensors_saved_for_backwards))
```

Comment:
@Chillee this should now be a bit less egregious, and still fixes the mem leak 🙂 (we only need to worry about the case when our saved-for-bw tensor is a differentiable view)